themes-default/slim/views/vue-components/app-link.mako

```diff
@@ -22,80 +24,103 @@ Vue.component('app-link', {
         placeholder: {
             type: String,
             default: 'indexer-to-name'
-        }
+        },
+        base: String
     },
     computed: {
+        config() {
+            return this.$store.state.config;
+        },
+        indexerName() {
+            const { indexers } = config.indexers.config;
+            // Returns `undefined` if not found
+            return Object.keys(indexers).find(indexer => indexers[indexer].id === parseInt(indexerId, 10));
+        },
+        computedBase() {
+            // Return prop before HTML element to allow tests to mock
+            if (this.base) {
+                return this.base;
+            }
+
+            return document.getElementsByTagName('base')[0].getAttribute('href');
+        },
+        computedHref() {
+            const { href, indexerId, placeholder, indexerName } = this;
+            if (indexerId && placeholder) {
+                return href.replace(placeholder, indexerName);
+            }
+            return href;
+        },
+        isIRC() {
+            return this.computedHref.startsWith('irc://');
+        },
+        isAbsolute() {
+            const href = this.computedHref;
+            return /^[a-z][a-z0-9+.-]*:/.test(href);
+        },
+        isExternal() {
+            const base = this.computedBase;
+            const href = this.computedHref;
+            return !href.startsWith(base) && !href.startsWith('webcal://');
+        },
+        isHashPath() {
+            return this.computedHref.startsWith('#');
+        },
+        anonymisedHref() {
+            const { anonRedirect } = this.config;
+            const href = this.computedHref;
+            return anonRedirect ? anonRedirect + href : href;
+        },
         linkProperties() {
-            const {to, indexerId, placeholder} = this;
-            const href = indexerId && placeholder ? this.href.replace(placeholder, MEDUSA.config.indexers.indexerIdToName(indexerId)) : this.href;
-            const base = document.getElementsByTagName('base')[0].getAttribute('href');
-            const isIRC = url => {
-                return url.startsWith('irc://');
-            };
-            const isAbsolute = url => {
-                return /^[a-z][a-z0-9+.-]*:/.test(url);
+            const { to, isIRC, isAbsolute, isExternal, isHashPath, isAnonymised, anonymisedHref } = this;
+            const base = this.computedBase;
+            const href = this.computedHref;
+
+            // Return normal router-link
+            if (to) {
+                return {
+                    is: 'router-link',
+                    to: { name: href }
```

Comment:
Side note...
Have you tried it? This technique doesn't seem to actually work.

Reply:
How did you test this? The `to` needs to be the name of a loaded route.

Reply:
When on the help and info page:
```vue
<app-link to="not-found">Not Found</app-link>
<app-link to="config">Help &amp; Info</app-link>
```
And the elements I got in the DOM are:
```html
<a class="router-link-exact-active router-link-active">Not Found</a>
<a class="router-link-exact-active router-link-active">Help &amp; Info</a>
```

It doesn't work even when I put the links inside the routes, like in `config.vue` and `404.vue`.

Reply:
`<router-link>` in the same locations, with the same props, works just fine.

Reply:
I think we should replace `to: { name: href }` with `to`?

Reply:
> Shall we merge and then fix this in the test PR?

No, the app link component is broken on this branch.
Give me a few minutes I'll push a commit to fix everything.

Reply:
I'll try and fix this another time. Fixed the other errors.